Added AutoChargeStationSequence to drive to charging station and call…

avakimc opened the pull request:
… balancing command

avakimc committed:
Added AutoChargeStationSequence to drive to charging station and call balancing command

avakimc committed:
Fixed AutoChargeStationSequence Command

avakimc committed:
Revisions to be able to build

balghane commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -40,4 +46,8 @@ public RobotContainer() {
   }
+
+  public static Command getAutonomousCommand(){
+    return new AutoChargeStationSequence(true, new DriveSubsystem());
```
Can you make `true` a named variable in this function just so people don't have to go into AutoChargeStationSequence to see what the true is referring to?

balghane commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -40,4 +46,8 @@ public RobotContainer() {
   }
+
+  public static Command getAutonomousCommand(){
+    return new AutoChargeStationSequence(true, new DriveSubsystem());
```
Can you instantiate a driveSubsystem as a member variable in this class, and then just pass that instance into this function?

avakimc replied:
Okay, when I did this I also changed the method so it's not longer static.

balghane commented on src/main/java/frc/robot/commands/AutoChargeStationSequence.java:
```diff
@@ -0,0 +1,28 @@
+import frc.robot.Calibrations;
+import frc.robot.subsystems.drive.DriveSubsystem;
+
+public class AutoChargeStationSequence extends SequentialCommandGroup {
```
Can you add a top-level `/** */` style comment for what this auto does

avakimc committed:
Revisions to AutoChargeStationBalance

- Added a comment describing what the autonomous does
- Changed the getAutonomousCommand() method so that it's not longer static, and calls it on an instance of RobotContainer in Robot.java
- Made boolean isFirstPath in the method getAutonomousCommand() and made drive a member variable in RobotContainer

balghane commented:
/format

github-actions[bot] committed:
Formatting fixes

balghane commented on src/main/java/frc/robot/Calibrations.java:
```diff
@@ -72,4 +72,14 @@ public static final class SwerveModuleCalibrations {
       INTAKE_DEPLOYED_POSITION_DEGREES = PLACEHOLDER_DOUBLE;
+
+  /** Time to drive the robot to approach the charging station in auto in seconds */
+  public static final double DRIVE_TIME_AUTO_SECONDS = PLACEHOLDER_DOUBLE;
```
We can remove this one

avakimc replied:
OK

avakimc committed:
Deleted DRIVE_TIME_AUTO_SECONDS Variable in Calibrations
